src/pow.rs

```diff
@@ -0,0 +1,1540 @@
+        arr[0] == 0 && arr[1] == 0
+    }
+
+    fn is_one(&self) -> bool {
+        let Uint256(ref arr) = self;
+        arr[0] == 0 && arr[1] == 1
+    }
+
+    fn is_max(&self) -> bool {
+        let Uint256(ref arr) = self;
+        arr[0] == u128::max_value() && arr[1] == u128::max_value()
+    }
+
+    /// Returns the low 32 bits.
+    fn low_u32(&self) -> u32 { self.low_u128() as u32 }
+
+    /// Returns the low 64 bits.
+    fn low_u64(&self) -> u64 { self.low_u128() as u64 }
+
+    /// Returns the low 128 bits.
+    fn low_u128(&self) -> u128 {
+        let &Uint256(ref arr) = self;
+        arr[1]
+    }
+
+    /// Returns the least number of bits needed to represent the number.
+    fn bits(&self) -> u32 {
+        let &Uint256(ref arr) = self;
+        if arr[0] > 0 {
+            256 - arr[0].leading_zeros()
+        } else {
+            128 - arr[1].leading_zeros()
+        }
+    }
+
+    /// Wrapping multiplication by `u64`.
+    fn mul_u64(self, rhs: u64) -> Uint256 {
+        if self.is_zero() || rhs == 0 {
+            return Uint256::ZERO;
+        }
+
+        let Uint256(ref arr) = self;
+        let mut carry = [0u128; 2];
+        let mut ret = [0u128; 2];
+
+        for (i, word) in arr.iter().enumerate().rev() {
+            let not_last_word = i == 1;
+            let upper = (rhs as u128) * (word >> 64);
+            let lower = (rhs as u128) * (word & 0xFFFF_FFFF_FFFF_FFFF);
+            if not_last_word {
+                carry[0] += upper >> 64;
+            }
+            let (sum, overflow) = lower.overflowing_add(upper << 64);
+            ret[i] = sum;
+            if overflow && not_last_word {
+                carry[0] += 1;
+            }
+        }
+        Uint256(ret).wrapping_add(Uint256(carry))
+    }
+
+    /// Calculates quotient and remainder.
+    ///
+    /// # Returns
+    ///
+    /// (quotient, remainder)
+    ///
+    /// # Panics
+    ///
+    /// If `rhs` is zero.
+    fn div_rem(self, rhs: Self) -> (Self, Self) {
+        let mut sub_copy = self;
+        let mut shift_copy = rhs;
+        let mut ret = [0u128; 2];
+
+        let my_bits = self.bits();
+        let your_bits = rhs.bits();
+
+        // Check for division by 0
+        assert!(your_bits != 0, "attempted to divide {} by zero", self);
+
+        // Early return in case we are dividing by a larger number than us
+        if my_bits < your_bits {
+            return (Uint256(ret), sub_copy);
+        }
+
+        // Bitwise long division
+        let mut shift = my_bits - your_bits;
+        shift_copy = shift_copy << shift;
+        loop {
+            if sub_copy >= shift_copy {
+                ret[1 - (shift / 128) as usize] |= 1 << (shift % 128);
+                sub_copy = sub_copy.wrapping_sub(shift_copy);
+            }
+            shift_copy = shift_copy >> 1;
+            if shift == 0 {
+                break;
+            }
+            shift -= 1;
+        }
+
+        (Uint256(ret), sub_copy)
```

Comment:
Oh, crap now that I wrote all the ideas how to make it `const fn` I realized I don't see how to fix this. Maybe my math skills are not good enough or maybe it's impossible.

Reply:
I had a go at const'ing loads of things before. I don't think its worth the added complexity (and effort) at the moment. Rust 1.41 is too painful for widespread const'ing of functions in my opinion.